### ISSUE TITLE
Add dataset to log messages

### ISSUE THREAD

**Issue by stuarteberg** [connectome-neuprint/neuPrintHTTP#63]
Right now the log doesn't say which dataset is being accessed.

To allow us to continue using the same parsing logic, please make sure the new `dataset` field is NOT the last item in the log message.  Ideally, put it as the first key in the JSON.

Example:

```
{
"dataset": "hemibrain",        // <-- put it here
"uri": "/api/custom/custom",
"status": 200,
"bytes_in": 526,
"bytes_out": 1050222,
"duration": 63.75,
"time": 1697977509,
"user": "foobarfoo@gmail.com",
"category": "custom/custom",
"debug": "
    MATCH (m:Meta)
    WITH m as m,
         apoc.convert.fromJsonMap(m.roiInfo) as roiInfo,
         apoc.convert.fromJsonMap(m.roiHierarchy) as roiHierarchy,
         apoc.convert.fromJsonMap(m.neuroglancerInfo) as neuroglancerInfo,
         apoc.convert.fromJsonList(m.neuroglancerMeta) as neuroglancerMeta,
         apoc.convert.fromJsonMap(m.statusDefinitions) as statusDefinitions
    RETURN m as meta, roiInfo, roiHierarchy, neuroglancerInfo, neuroglancerMeta, statusDefinitions
"
// definitely NOT here.
}
```

**Comment by DocSavage** [comment]
Keep open until testing.

**Comment by neomorphic** [comment]
The changes made in 1d440bd don not quite resolve the issue as that updates the DefaultLoggerConfig. That configuration is overridden by the one in [main.go](https://github.com/connectome-neuprint/neuPrintHTTP/blob/437f0c421ea796fc837b7e720f8569644ab47496/main.go#L133). Also it looks like adding the "dataset":${dataset} section to that configuration is returning an empty string for all the requests that I tested.

**Comment by neomorphic** [comment]
It looks like the issue is probably down to the fact that `c.Bind(&req)` is called multiple times in a single request cycle. From what I can gather the first time that function is called it reads through the request body and leaves it in a state that can no longer be read. It looks like we would need to move to a more complicated parsing of the body using c.GetBody.

**Comment by DocSavage** [comment]
This is where unit tests would be very convenient to rapidly iterate to a solution that works.

**Comment by DocSavage** [comment]
It seems like [this `c.Bind` might be a mistake](https://github.com/connectome-neuprint/neuPrintHTTP/blob/1d440bd3e0ec065fa4d59193657ce071c233d7b9/logging/config.go#L121). The `req` is already available as it is in other case statements, right?

**Comment by DocSavage** [comment]
Dataset should now be logged for a majority of the requests. Obviously general requests (e.g., to get datasets, profile, token) will have `dataset` set to zero value which is empty string. This was fixed in two commits ending with commit dc00f72.
